syntax = "proto3";

package autosampler;

option go_package = "github.com/jt05610/petri/devices/autosampler";

service Autosampler {
  rpc Inject(InjectRequest) returns (stream InjectResponse);

}

message InjectRequest {
    int32 vial = 1;
    int32 air_cushion = 2;
    int32 excess_volume = 3;
    int32 flush_volume = 4;
    int32 injection_volume = 5;
    int32 needle_depth = 6;
}

enum InjectState {
  Idle = 0;
  Injecting = 1;
  Injected = 2;
}

message InjectResponse {
  InjectState state = 1;
}